common/src/main/java/org/keycloak/common/Profile.java

```diff
@@ -104,7 +104,7 @@ public enum Feature {
 
         MULTI_SITE("Multi-site support", Type.DISABLED_BY_DEFAULT),
 
-        REMOTE_CACHE("Remote caches support. Requires Multi-site support to be enabled as well.", Type.EXPERIMENTAL),
+        STATELESS("External Infinispan support. Requires Multi-site support to be enabled as well.", Type.EXPERIMENTAL),
```

Comment:
We should probably update this description to state that user sessions will be volatile if enabled.

Reply:
IMHO Actually not unless persistent sessions are disabled (which is now enabled by default)

Reply:
My mistake. Can we succinctly summarize what the difference between a multi-site deployment with/without the stateless feature flag is? At this stage I'm confused by the subtleties... is this documented anywhere?

Reply:
Looking at the code, should be like this:

* multi-site enabled -> external Infinispan cluster + sessions stored in database
* multi-site enabled, persistent-user-sessions disabled -> same as above (persistent is forced)
* multi-site enabled, remote-cache enabled, persistent-user-sessions disabled -> external Infinispan cluster +  volatile sessions
* remote-cache enabled -> single site with external Infinispan + sessions stored in database
* remote-cache enabled , persistent-user-sessions disabled -> single site with external Infinispan + volatile sessions

we are trying to find a new name for `remote-cache` feature.

Reply:
> * multi-site enabled, persistent-user-sessions disabled -> same as above (persistent is forced)
> 
> * multi-site enabled, remote-cache enabled, persistent-user-sessions disabled -> external Infinispan cluster +  volatile sessions

Shouldn't these too be the same? As a user if I have explicitly asked for `persistent-user-sessions` to be disabled it seems very counter-intuitive to me that I also need to add a "stateless/remote-cache" feature for this to actually be the case.

> remote-cache enabled -> single site with external Infinispan + sessions stored in database

If we're allowing this now, then the description in `Profile` needs to be updated. I'm guessing you're talking about future intent though.

TBH I thought we agreed to limit the number of feature combinations we were going to support. IMO less is more and we're introducing way too many combinations here, which while it may satisfy some power users, will ultimately will confuse the majority of users. I'm working on this full time and I can't keep up :upside_down_face:

Reply:
> Shouldn't these too be the same? As a user if I have explicitly asked for `persistent-user-sessions` to be disabled it seems very counter-intuitive to me that I also need to add a "stateless/remote-cache" feature for this to actually be the case.

"Volatile Sessions" is experimental at this point. Just disabling `persistent-user-sessions` is not good enough IMO. There is no indication that the configuration is not supported. Having a feature marked experimental that needs to be explicitly enabled makes more sense to me. 

> If we're allowing this now, then the description in `Profile` needs to be updated. I'm guessing you're talking about future intent though.
> 
> TBH I thought we agreed to limit the number of feature combinations we were going to support. IMO less is more and we're introducing way too many combinations here, which while it may satisfy some power users, will ultimately will confuse the majority of users. I'm working on this full time and I can't keep up 🙃

I chatted with Alexander and we agreed to make it experimental in KC 26. We want to support it in KC 26.x and it is easier to do if the feature already exists than create a new one during a minor release. 

The confusion would be worse when everything is supported. `multi-site` and `remote-cache`/`stateless` are redundant and Keycloak does not care if the external Infinispan cluster is deployed in a single-site or cross-site mode.

Reply:
> "Volatile Sessions" is experimental at this point. Just disabling `persistent-user-sessions` is not good enough IMO. There is no indication that the configuration is not supported. Having a feature marked experimental that needs to be explicitly enabled makes more sense to me.

That's a fair point. In that case, should we warn or even thrown an Exception when users configure `multi-site enabled, !persistent-user-sessions` ? If the behaviour is the same as just enabling `multi-site` feature, I believe configuring both options would expect persistent to be disabled, so we should make it clear that is NOT the case.


> I chatted with Alexander and we agreed to make it experimental in KC 26. We want to support it in KC 26.x and it is easier to do if the feature already exists than create a new one during a minor release.

Ok. I'm still not convinced giving users all these options in the long run is a good idea, multi-site is hard enough for users with a single feature, but here's not the place for that discussion :slightly_smiling_face: